src/main/java/com/minecolonies/coremod/blocks/BlockMinecoloniesRack.java

```diff
@@ -0,0 +1,383 @@
+    public IBlockState getActualState(final IBlockState state, final IBlockAccess worldIn, final BlockPos pos)
+    {
+        final TileEntity entity = worldIn.getTileEntity(pos);
+
+        if (!(entity instanceof TileEntityRack))
+        {
+            return super.getActualState(state, worldIn, pos);
+        }
+
+        final TileEntityRack rack = (TileEntityRack) entity;
+        if (rack.isEmpty() && (rack.getOtherChest() == null || rack.getOtherChest().isEmpty()))
+        {
+            if (rack.getOtherChest() != null)
+            {
+                if (rack.isMain())
+                {
+                    return state.withProperty(BlockMinecoloniesRack.VARIANT, RackType.DEFAULTDOUBLE)
+                             .withProperty(FACING, BlockPosUtil.getFacing(rack.getNeighbor(), pos));
+                }
+                else
+                {
+                    return state.withProperty(BlockMinecoloniesRack.VARIANT, RackType.EMPTYAIR);
+                }
+            }
+            else
+            {
+                return state.withProperty(BlockMinecoloniesRack.VARIANT, RackType.DEFAULT);
+            }
+        }
+        else
+        {
+            if (rack.getOtherChest() != null)
+            {
+                if (rack.isMain())
+                {
+                    return state.withProperty(BlockMinecoloniesRack.VARIANT, RackType.FULLDOUBLE)
+                             .withProperty(FACING, BlockPosUtil.getFacing(rack.getNeighbor(), pos));
+                }
+                else
+                {
+                    return state.withProperty(BlockMinecoloniesRack.VARIANT, RackType.EMPTYAIR);
+                }
+            }
+            else
+            {
+                return state.withProperty(BlockMinecoloniesRack.VARIANT, RackType.FULL);
+            }
+        }
+    }
+
+    /**
+     * Convert the BlockState into the correct metadata value.
+     *
+     * @deprecated (Remove this as soon as minecraft offers anything better).
+     */
+    @NotNull
+    @Override
+    @Deprecated
+    public IBlockState withRotation(@NotNull final IBlockState state, final Rotation rot)
+    {
+        return state.withProperty(FACING, rot.rotate(state.getValue(FACING)));
+    }
+
+    /**
+     * @deprecated (Remove this as soon as minecraft offers anything better).
+     */
+    @NotNull
+    @Override
+    @Deprecated
+    public IBlockState withMirror(@NotNull final IBlockState state, final Mirror mirrorIn)
+    {
+        return state.withRotation(mirrorIn.toRotation(state.getValue(FACING)));
+    }
+
+    /**
+     * @deprecated (Remove this as soon as minecraft offers anything better).
+     */
+    @Override
+    @Deprecated
+    public boolean isFullCube(final IBlockState state)
+    {
+        return false;
+    }
+
+    /**
+     * @deprecated (Remove this as soon as minecraft offers anything better).
+     */
+    @Override
+    @Deprecated
+    public boolean isOpaqueCube(final IBlockState state)
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Do not forget to remove this deprecated code someday. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1133)